client/src/partials/budget/new/new_budget.js

```diff
@@ -0,0 +1,365 @@
+angular.module('bhima.controllers')
+.controller('NewBudgetController', NewBudgetController);
+
+NewBudgetController.$inject = [
+  '$q', 
+  '$scope', 
+  '$http', 
+  '$translate', 
+  'validate', 
+  'precision', 
+  'messenger', 
+  'SessionService', 
+  'connect', 
+  'Upload'
+];
+
+/**
+*  New Budget Controller
+*  This controller is responsible for creating a new budget by importing a csv file or individually by account
+*  The csv file imported is is a file which already has been processed by an accountant
+*/
+function NewBudgetController($q, $scope, $http, $translate, validate, precision, messenger, SessionService, connect, Upload) {
+  var dependencies = {},
+      enterprise_id = null,
+      session = $scope.session = {},
+      config = $scope.config = {};
+
+  // Set up for the database queries
+  dependencies.account = {
+    query : {
+      tables : {
+        'account' :{
+          columns : ['id', 'account_type_id', 'account_txt', 'account_number']
+        },
+        'account_type' : { 
+          columns : ['type'] 
+        },
+      },
+      join : ['account_type.id = account.account_type_id'],
+      where : [ 'account.account_type_id in (1,4)' ]
+    }
+  };
+
+  dependencies.budgets = {
+    query : {
+    tables : {
+      'budget' : { 
+        columns : ['id', 'account_id', 'period_id', 'budget']
+      },
+      'period' : {
+        columns : ['fiscal_year_id', 'period_number', 'period_start', 'period_stop', 'locked' ]
+      }
+    },
+    join  : [ 'period.id=budget.period_id' ],
+    where : [ 'period.period_number<>0' ]
+    }
+  };
+
+  dependencies.periods = {
```

Comment:
This is a duplication with the function loadPeriod.  In the loadPeriod function (line [137](https://github.com/IMA-WorldHealth/bhima/pull/776/files#diff-026166d2e4e2dd47428c20c9eccf2c53R137)) you set up `dependencies.period` (note: period !== periods, but they do very close to the same thing).  If you could combine the two requests (dependencies.period and dependencies.periods) in to the same request object, that would be much cleaner.